src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -141,11 +142,49 @@ protected string ParseServiceName(ConfigurationKeyValue kv)
 			Logger?.Error()
 				?.Log("Failed calculating service name, the service name will be 'unknown'." +
 					" You can fix this by setting the service name to a specific value (e.g. by using the environment variable {ServiceNameVariable})",
-					ConfigConsts.ConfigKeys.ServiceName);
+					ConfigConsts.EnvVarNames.ServiceName);
 
 			return "unknown";
 		}
 
+		protected double ParseTransactionSampleRate(ConfigurationKeyValue kv)
+		{
+			if (kv?.Value == null)
+			{
+				Logger?.Debug()
+					?.Log("No transaction sample rate provided. Defaulting to '{DefaultTransactionSampleRate}'",
+						ConfigConsts.DefaultValues.TransactionSampleRate);
+				return ConfigConsts.DefaultValues.TransactionSampleRate;
+			}
+
+			if (!double.TryParse(kv.Value, out var parsedValue))
```

Comment:
This will be dependent on the culture settings of the machine (`0.5` vs `0,5`) and I think that can be very dangerous. Should we maybe pin it to a specific culture and document it accordingly? 

E.g.: 
`double.TryParse(kv.Value,  System.Globalization.NumberStyles.Any, CultureInfo.GetCultureInfo("en-US"), out var parsedValue)`.

Also, maybe I overlooked it, but I think this is not covered by any test. It'd be useful to have a test in the `EnvVarConfigTest` for this. You can simply set the env. var and make sure the it's parsed correctly. E.g. a test that tries both `0.5` and `0,5`, and make sure always the correct one wins.

Reply:
Done. But just to make sure I understood you correctly - you meant to try and parse configuration value set via environment variable but you didn't mean to use environment variable to set some other (not "en-US") culture. Did I understand you correctly? BTW I don't know if it's even possible to change application's current culture and if it's setting which environment variable will achieve that.

Reply:
Yes, this code is in my opinion safe now. 👍  What I meant is that for example with my culture setting (you can simulate it by using `CultureInfo.GetCultureInfo("de-at")` the `TryParse` method would expect '`,`' as the separator and not `'.`'. So e.g. on my machine I'd need to use `,` to set this value, but once I deploy let's see to a server with `en-us` local then suddenly the line with thw `TryParse` would expect '`,`'  as the separator. By using ` CultureInfo.InvariantCulture` this is now fixed to '`.`' on every machine. So we don't change the region setting of the machine, we just tell `TryParse` how to interpret the numbers.